src/Microsoft.Extensions.Logging.Abstractions/EventId.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.Extensions.Logging
+{
+    public struct EventId
+    {
+        private int _id;
+        private string _name;
+
+        public EventId(int id, string name = null)
+        {
+            _id = id;
+            _name = name;
+        }
+
+        public static implicit operator int(EventId e)
+        {
+            return e._id;
+        }
+
+        public static implicit operator EventId(int i)
+        {
+            return new EventId(i);
+        }
+
+        public static explicit operator string(EventId e)
```

Comment:
Should have Id and Name get-only properties. Also to @Tratcher's question, we might not need the implicit operators for the int/string values.